docs/design/signature-provider-plugins.md

```diff
@@ -0,0 +1,444 @@
+       <PackageReference Include="Azure.Security.KeyVault.Keys" Version="4.2.0" PrivateAssets="all" />
+       <PackageReference Include="RSAKeyVaultProvider" Version="2.1.1" PrivateAssets="all" />
+   </ItemGroup>
+   ```
+
+4. Add public implementations for relevant interfaces defined in the interfaces package.
+5. Add a [`plugin.json`](#plugin-json-file) file to the project.
+6. Update the plugin's project file to create a NuGet package.
+
+   ```XML
+   <PropertyGroup>
+       <!-- Enable NuGet package creation. -->
+       <IsPackable>true</IsPackable>
+
+       <!--
+       Be sure to add other mandatory and optional properties for creating a NuGet package.
+       See https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#pack-target.
+       -->
+
+       <!-- Enable rolling forward to a later runtime. -->
+       <!-- See https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#rollforward -->
+       <RollForward>LatestMajor</RollForward>
+
+       <!-- Copy runtime dependencies to the build output directory. -->
+       <!-- https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#enabledynamicloading -->
+       <EnableDynamicLoading>true</EnableDynamicLoading>
+
+       <!-- All private dependencies must be internalized.  Enable Sign CLI to load private dependencies using the plugin's deps.json file. -->
+       <GenerateDependencyFile>true</GenerateDependencyFile>
+       <TargetsForTfmSpecificBuildOutput>$(TargetsForTfmSpecificBuildOutput);CopyProjectReferencesToPackage</TargetsForTfmSpecificBuildOutput>
+   </PropertyGroup>
+   
+   <ItemGroup>
+       <!-- This adds plugin.json to the root directory of the plugin's package. -->
+       <Content Include="plugin.json">
+           <Pack>true</Pack>
+           <!-- An empty value means the root directory. -->
+           <PackagePath></PackagePath>
+       </Content>
+   </ItemGroup>
+   
+   <!-- This target copies runtime dependencies to the build output directory. -->
+   <Target Name="CopyProjectReferencesToPackage" DependsOnTargets="ResolveReferences">
+       <ItemGroup>
+           <BuildOutputInPackage
+               Include="@(ReferenceCopyLocalPaths)"
+               TargetPath="%(ReferenceCopyLocalPaths.DestinationSubPath)" />
+       </ItemGroup>
+   </Target>
+   
+   <!-- This target adds the generated deps.json file to the build output directory. -->
+   <Target Name="AddBuildDependencyFileToBuiltProjectOutputGroupOutput"
+           BeforeTargets="BuiltProjectOutputGroup"
+           Condition=" '$(GenerateDependencyFile)' == 'true'">
+   
+       <ItemGroup>
+           <BuiltProjectOutputGroupOutput
+               Include="$(ProjectDepsFilePath)"
+               TargetPath="$(ProjectDepsFileName)"
+               FinalOutputPath="$(ProjectDepsFilePath)" />
+       </ItemGroup>
+   </Target>
+   ```
+
+#### <a name="plugin-json-file"></a>The `plugin.json` file
```

Comment:
I wonder if the design can be simplified by moving the plugin.json content elsewhere. For example: name can be inferred from package name or from class name, or it can be part of the ISignatureProviderPlugin. Path can be inferred from looking at all the classes that implement ISignatureProviderPlugin and auto-adding them. 
The reason I'm suggesting this, is because editing an additional file can be error prone for developers and increase plugin development time.